### ISSUE TITLE
Allow void async functions with "return-undefined"

### ISSUE THREAD

**Issue by Kovensky** [palantir/tslint#2565]
### Bug Report

- __TSLint version__: 5.1.0
- __TypeScript version__: 2.3.0
- __Running TSLint via__: webpack/tslint-loader

#### TypeScript code being linted

```ts
async function foo (bar: boolean) {
  if (!bar) {
    return
  }
}
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint-config-standard"
  ]
}
```

#### Actual behavior

`return-undefined` complains about the `return` not being a `return undefined`

#### Expected behavior

`return-undefined` should allow the `return` to stand, otherwise it is not possible to naturally write a void `async` function with early returns.
